src/providers/Subgraph.js

```diff
@@ -0,0 +1,43 @@
+import React from 'react'
+import {
+  createClient,
+  Provider as UrqlProvider,
+  cacheExchange,
+  debugExchange,
+} from 'urql'
+import { getFetchExchange, getSubscriptionExchange } from '../graphql-exchanges'
+
+import { devtoolsExchange } from '@urql/devtools'
+import { defaultSubgraphHttpEndpoint } from '../networks'
+
+const SubgraphContext = React.createContext({ resetSubgraphClient: null })
+
+const newClient = resetSubscription =>
+  createClient({
+    url: defaultSubgraphHttpEndpoint,
+    exchanges: [
+      debugExchange,
+      devtoolsExchange,
+      cacheExchange,
+      getFetchExchange(),
+      getSubscriptionExchange(resetSubscription),
+    ],
+  })
+
+function SubGraphProvider({ children }) {
+  const [client, setClient] = React.useState(newClient(false))
+
+  return (
+    <SubgraphContext.Provider
+      value={{
+        resetSubgraphClient: () => setClient(newClient(true)),
+      }}
+    >
+      <UrqlProvider value={client}>{children}</UrqlProvider>
+    </SubgraphContext.Provider>
+  )
+}
```

Comment:
Since we are reloading the window on every network change, is this provider really needed ?

Reply:
ahh i though that would be better to have an specific provider for that with an specific action instead of just relying on the web page reload, is there any specific reason why you want to remove it?

Reply:
Yes i agree that having it encapsulated in a component looks cleaner, just was thinking that maybe the logic of resetting it and saving the client as a state wasn't needed because we are reloading the page any ways and i don't think we'll be resetting it from anywhere else except from the Network settings

Reply:
Coming back to this, these lines definitely seem like they aren't necessary since we're just refreshing the page later:

https://github.com/aragon/court-dashboard/pull/277/files#diff-9eff867218c8af415bc22e0835ba9dd7R181-R183